bin/generated_template.d.ts

```diff
@@ -0,0 +1,113 @@
+/** Provides information pointing to a location within a source. */
+export interface Location {
+  /** Line in the parsed source (1-based). */
+  line: number;
+  /** Column in the parsed source (1-based). */
+  column: number;
+  /** Offset in the parsed source (0-based). */
+  offset: number;
+}
+
+/** The `start` and `end` position's of an object within the source. */
+export interface LocationRange {
+  /** Any object that was supplied to the `parse()` call as the `grammarSource` option. */
+  source: any;
+  /** Position at the beginning of the expression. */
+  start: Location;
+  /** Position after the end of the expression. */
+  end: Location;
+}
+
+export interface LiteralExpectation {
+  type: "literal";
+  text: string;
+  ignoreCase: boolean;
+}
+export interface ClassParts extends Array<string | ClassParts> {
+}
+export interface ClassExpectation {
+  type: "class";
+  parts: ClassParts;
+  inverted: boolean;
+  ignoreCase: boolean;
+}
+export interface AnyExpectation {
+  type: "any";
+}
+export interface EndExpectation {
+  type: "end";
+}
+export interface OtherExpectation {
+  type: "other";
+  description: string;
+}
+export type Expectation =
+  | AnyExpectation
+  | ClassExpectation
+  | EndExpectation
+  | LiteralExpectation
+  | OtherExpectation;
+
+declare class _PeggySyntaxError extends Error {
```

Comment:
You can write `export declare class SyntaxError extends Error {` and delete the line `export declare const SyntaxError: typeof _PeggySyntaxError;` and `export type SyntaxError = _PeggySyntaxError;`. They should be equivalent as you can import a class as both a type and a value.

Yes Peggy emits function-like classes but hiding it behind a variable doesn't remove any class properties which is fortunate actually.

Reply:
I can't mark this as resolved but it can be.